tests/integration/scoring/test_scoring.py

```diff
@@ -106,47 +162,60 @@ def test_scoring_score_with_params_llm_as_judge(llama_stack_client, sample_judge
         assert len(response.results[x].score_rows) == 5
 
 
-@pytest.mark.skip(reason="Skipping because this seems to be really slow")
-def test_scoring_score_with_aggregation_functions(llama_stack_client, sample_judge_prompt_template, judge_model_id):
+@pytest.mark.parametrize(
+    "provider_id",
+    [
+        "basic",
+        "llm-as-judge",
+        "braintrust",
+    ],
+)
+def test_scoring_score_with_aggregation_functions(
+    llama_stack_client, sample_judge_prompt_template, judge_model_id, provider_id
+):
     register_dataset(llama_stack_client, for_rag=True)
     rows = llama_stack_client.datasetio.get_rows_paginated(
         dataset_id="test_dataset",
         rows_in_page=3,
     )
     assert len(rows.rows) == 3
 
-    scoring_fns_list = llama_stack_client.scoring_functions.list()
+    scoring_fns_list = [x for x in llama_stack_client.scoring_functions.list() if x.provider_id == provider_id]
+    if len(scoring_fns_list) == 0:
+        pytest.skip(f"No scoring functions found for provider {provider_id}, skipping")
+
     scoring_functions = {}
     aggr_fns = [
         "accuracy",
         "median",
         "categorical_count",
         "average",
     ]
-    for x in scoring_fns_list:
-        if x.provider_id == "llm-as-judge":
-            aggr_fns = ["categorical_count"]
-            scoring_functions[x.identifier] = dict(
-                type="llm_as_judge",
-                judge_model=judge_model_id,
-                prompt_template=sample_judge_prompt_template,
-                judge_score_regexes=[r"Score: (\d+)"],
+
+    scoring_fn = scoring_fns_list[0]
```

Comment:
why the first one?

Reply:
We test 1 scoring function per provider, as `braintrust` has 10+ scoring functions, and its slow to loop over all.